ext/soap/soap.c

```diff
@@ -983,6 +983,11 @@ PHP_METHOD(SoapServer, __construct)
 			}
 		}
 
+		if ((tmp = zend_hash_find(ht, ZSTR_KNOWN(ZEND_STR_TRACE))) != NULL &&
+		    (Z_TYPE_P(tmp) == IS_TRUE ||
+		     (Z_TYPE_P(tmp) == IS_LONG && Z_LVAL_P(tmp) == 1))) {
```

Comment:
nit: you do not want to treat as string literals eventually ?

Reply:
I'm not sure I understand the question.
In any case, this check was copied over from SoapClient so that the behaviour is consistent.

Reply:
I meant considering "trace" => "true" but if this is consistent then it s fine :)

Reply:
Oh I see. Some people may be tempted to do this, and the annoying thing is that passing wrong types in the options array does not cause an exception, the option just gets ignored in that case... however, instead of allowing "true" I rather start throwing exceptions on wrong options.